chain/xrp/client/client.go

```diff
@@ -0,0 +1,906 @@
+	}
+
+	return &txInput, nil
+}
+
+// Deprecated method - use FetchTransferInput
+func (client *Client) FetchLegacyTxInput(ctx context.Context, from xc.Address, to xc.Address) (xc.TxInput, error) {
+	// No way to pass the amount in the input using legacy interface, so we estimate using min amount.
+	args, _ := xcbuilder.NewTransferArgs(from, to, xc.NewAmountBlockchainFromUint64(1))
+	txInput, err := client.FetchTransferInput(ctx, args)
+	if err != nil {
+		return nil, err
+	}
+
+	return txInput, nil
+}
+
+// SubmitTx submits a Template tx
+func (client *Client) SubmitTx(ctx context.Context, txInput xc.Tx) error {
+	serializedTxInputBytes, err := txInput.Serialize()
+	if err != nil {
+		return err
+	}
+
+	serializedTxInputHex := hex.EncodeToString(serializedTxInputBytes)
+	serializedTxInputHexBytes := []byte(serializedTxInputHex)
+
+	submitRequest := &SubmitRequest{
+		Method: "submit",
+		Params: []SubmitParamEntry{
+			{
+				TxBlob: string(serializedTxInputHexBytes),
+			},
+		},
+	}
+
+	var submitResponse SubmitResponse
+	err = client.Send(MethodPost, submitRequest, &submitResponse)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// FetchTxInfo Returns transaction info - new endpoint
+func (client *Client) FetchTxInfo(ctx context.Context, txHash xc.TxHash) (xclient.TxInfo, error) {
+	legacyTxInfo, err := client.FetchLegacyTxInfo(ctx, txHash)
+	if err != nil {
+		return xclient.TxInfo{}, err
+	}
+
+	// Remap to new tx
+	return xclient.TxInfoFromLegacy(client.Asset.GetChain().Chain, legacyTxInfo, xclient.Account), nil
+}
+
+// FetchLegacyTxInfo Returns transaction info - legacy/old endpoint
+func (client *Client) FetchLegacyTxInfo(ctx context.Context, txHash xc.TxHash) (xc.LegacyTxInfo, error) {
+
+	txRequest := &TransactionRequest{
+		Method: "tx",
+		Params: []TransactionParamEntry{
+			{
+				Transaction: txHash,
+				Binary:      false,
+			},
+		},
+	}
+
+	var txResponse TransactionResponse
+	err := client.Send(MethodPost, txRequest, &txResponse)
+	if err != nil {
+		return xc.LegacyTxInfo{}, err
+	}
+
+	ledgerRequest := LedgerRequest{
+		Method: "ledger",
+		Params: []LedgerParamEntry{
+			{
+				LedgerIndex: "current",
+			},
+		},
+	}
+
+	var ledgerResponse LedgerResponse
+	err = client.Send(MethodPost, ledgerRequest, &ledgerResponse)
+	if err != nil {
+		return xc.LegacyTxInfo{}, err
+	}
+
+	confirmations := ledgerResponse.Result.LedgerCurrentIndex - txResponse.Result.Sequence
+
+	explorer := client.Asset.GetChain().ExplorerURL + "/tx/" + txResponse.Result.Hash + "?cluster=" + client.Asset.GetChain().Net
+
+	var sources []*xc.LegacyTxInfoEndpoint
+	var destinations []*xc.LegacyTxInfoEndpoint
+
+	if txResponse.Result.Destination != "" {
+		addSourceDestinationForPayment(txResponse, &sources, &destinations)
+	} else {
+		if txResponse.Result.TakerGets != nil {
+			handleTakerGets(txResponse, &sources, &destinations)
+		}
+
+		if txResponse.Result.TakerPays != nil {
+			handleTakerPays(txResponse, &sources, &destinations)
+		}
+	}
```

Comment:
I believe somewhere this does the "hard work" of parsing and understanding the tx metadata directly.  I wonder, if we just always do that, and ignore `Destination`, `TakerGets`, `TakerPays`, would it still work?

Reply:
Shouldn't the tx metadata still be accurate in all cases?